core/models/asset.py

```diff
@@ -576,16 +576,43 @@ def _save_initial_incident_report_state(self):
         existing_state = StateTransition.objects.filter(asset_incident_report=self)
         if not existing_state:
             StateTransition.objects.create(
-                asset_incident_report=self, state=constants.NEWLY_REPORTED
+                asset_incident_report=self,
+                incident_report_state=constants.NEWLY_REPORTED,
             )
-            self.save()
+
+        # user updates asset_status if lost
+        if self.incident_type == "Loss":
+            asset_status = AssetStatus.objects.filter(asset_id=self.asset.id).first()
+            asset_status.current_status = "Lost"
+            asset_status.save()
 
 
 class StateTransition(models.Model):
     asset_incident_report = models.ForeignKey(
         'AssetIncidentReport', on_delete=models.PROTECT
     )
-    state = models.CharField(max_length=50, default=constants.NEWLY_REPORTED)
+    incident_report_state = models.CharField(
+        max_length=50,
+        choices=constants.REPORT_STATE_OPTIONS,
+        default=constants.NEWLY_REPORTED,
+    )
+    asset_state = models.CharField(
+        max_length=50,
+        choices=constants.ASSET_STATE_OPTIONS,
+        default=constants.REQUIRES_REPAIR,
+    )
 
     class Meta:
         verbose_name_plural = 'State Transitions'
+
+    def save(self, *args, **kwargs):
+
+        # admin updates asset_status if damaged
+        if self.asset_state == "damaged":
+            asset_status = AssetStatus.objects.filter(
```

Comment:
we should be creating a new asset status here instead of updating the existing one. we normally do this for historical data.

Reply:
The implementation of the AssetStatus model is such that the previous status is saved once this code is hit. So it performs the update but keeps the previous status as well as can be shown in the screenshot.
<img width="708" alt="Screenshot 2019-05-09 at 14 41 25" src="https://user-images.githubusercontent.com/40039818/57450966-9b595900-7268-11e9-8a7b-c19a4d7f003d.png">

Reply:
this actually happens when a new status is added. if you add a `damaged` state the previous state will also be saved.

Reply:
see [this](https://github.com/AndelaOSP/art-backend/blob/ab3a9ed77f0a92cdc9cf870956bce193dd28e768/core/models/asset.py#L396)

Reply:
Yes, so I am not sure if I am understanding your point. There is a new status being created here. It replaces the old one which is saved as previous_status.

Reply:
we should just do `AssetStatus.objects.create(current_status=constants.DAMAGED, asset_id=self.asset_incident_report.asset_id)` instead of filtering the last added status